File: frontend/lib/data/repositories/comment_repository_impl.dart
import '../../domain/entities/comment.dart';
import '../../domain/repositories/comment_repository.dart';
import '../datasources/comment_remote_data_source.dart';

class CommentRepositoryImpl implements CommentRepository {
  final CommentRemoteDataSource remoteDataSource;

  CommentRepositoryImpl({
    required this.remoteDataSource,
  });

  @override
  Future<List<Comment>> getNovelComments(String novelId) async {
    try {
      return await remoteDataSource.getNovelComments(novelId);
    } catch (e) {
      throw Exception('Failed to get novel comments: $e');
    }
  }

  @override
  Future<List<Comment>> getChapterComments(String novelId, String chapterId) async {
    try {
      return await remoteDataSource.getChapterComments(novelId, chapterId);
    } catch (e) {
      throw Exception('Failed to get chapter comments: $e');
    }
  }

  @override
  Future<Comment> createComment(String novelId, String? chapterId, String content) async {
    try {
      return await remoteDataSource.createComment(novelId, chapterId, content);
    } catch (e) {
      throw Exception('Failed to create comment: $e');
    }
  }

  @override
  Future<Comment> updateComment(String novelId, String commentId, String content) async {
    try {
      return await remoteDataSource.updateComment(novelId, commentId, content);
    } catch (e) {
      throw Exception('Failed to update comment: $e');
    }
  }

  @override
  Future<void> deleteComment(String novelId, String commentId) async {
    try {
      await remoteDataSource.deleteComment(novelId, commentId);
    } catch (e) {
      throw Exception('Failed to delete comment: $e');
    }
  }
} 

File: adminapp/src/data/repositories/CommentRepositoryImpl.ts
import axiosInstance from '../../utils/axios';
import { Comment } from '../../domain/entities/Comment';
import { CommentRepository } from '../../domain/repositories/CommentRepository';

export class CommentRepositoryImpl implements CommentRepository {
  async getComments(): Promise<Comment[]> {
    const response = await axiosInstance.get<Comment[]>('/comments');
    return response.data;
  }

  async getCommentById(id: string): Promise<Comment> {
    const response = await axiosInstance.get<Comment>(`/comments/${id}`);
    return response.data;
  }

  async getCommentsByNovelId(novelId: string): Promise<Comment[]> {
    const response = await axiosInstance.get<Comment[]>(`/comments/novel/${novelId}`);
    return response.data;
  }

  async createComment(comment: Omit<Comment, 'id' | 'createdAt' | 'updatedAt'>): Promise<Comment> {
    const response = await axiosInstance.post<Comment>('/comments', comment);
    return response.data;
  }

  async updateComment(id: string, comment: Partial<Comment>): Promise<Comment> {
    const response = await axiosInstance.put<Comment>(`/comments/${id}`, comment);
    return response.data;
  }

  async deleteComment(id: string): Promise<void> {
    await axiosInstance.delete(`/comments/${id}`);
  }

  async getReportedComments(): Promise<Comment[]> {
    const response = await axiosInstance.get<Comment[]>('/comments/reported');
    return response.data;
  }

  async getCommentStats(): Promise<{
    totalComments: number;
    reportedComments: number;
    totalLikes: number;
  }> {
    const response = await axiosInstance.get<{
      totalComments: number;
      reportedComments: number;
      totalLikes: number;
    }>('/comments/stats');
    return response.data;
  }
} 